src/ax/persistence/messagestore.go

```diff
@@ -0,0 +1,43 @@
+package persistence
+
+import (
+	"context"
+
+	"github.com/jmalloc/ax/src/ax"
+)
+
+// MessageStore is an interface for persisting streams of messages.
+type MessageStore interface {
+	// AppendMessages appends one or more messages to a named stream.
+	//
+	// offset is a zero-based index into the stream. An error is returned if
+	// offset is not the next unused offset in the stream.
+	AppendMessages(
+		ctx context.Context,
+		tx Tx,
+		stream string,
+		offset uint64,
+		envs []ax.Envelope,
+	) error
+
+	// OpenStream opens a stream of messages for reading from a specific offset.
+	OpenStream(
```

Comment:
Document possible error conditions such as opening from an out of range offset, or stream not existing.